Origin: https://code.launchpad.net/~bcsaller/juju/log-output/+merge/103762
Author: Benjamin Saller <bcsaller@gmail.com>
Bug: http://pad.lv/915506
Description:
  Merge juju-log output fix [r=hazmat] [f=915506]
 
  juju-log in hooks uses the logging channels for IO and the command
  shouldn't output directly to stdout except in case of errors doing
  proper logging

Use --include-merged or -n0 to see merged revisions.
=== modified file 'juju/hooks/commands.py'
--- old/juju/hooks/commands.py	2012-04-24 21:47:40 +0000
+++ new/juju/hooks/commands.py	2012-05-03 16:46:40 +0000
@@ -137,7 +137,7 @@
 
     def run(self):
         return self.client.list_relations(self.options.client_id,
-                                          self.options.relation_id)
+                                   self.options.relation_id)
 
     def format_eval(self, result, stream):
         """ eval `juju-list` """
@@ -169,6 +169,9 @@
         return self.client.log(self.options.l,
                                self.options.message)
 
+    def render(self, result):
+        return None
+
 
 def log():
     """Entry point for juju-log."""

=== modified file 'juju/hooks/tests/test_invoker.py'
--- old/juju/hooks/tests/test_invoker.py	2012-04-25 00:20:26 +0000
+++ new/juju/hooks/tests/test_invoker.py	2012-05-02 00:49:10 +0000
@@ -33,8 +33,8 @@
         self.client = client
         self.socket_path = socket_path
         self.charm_dir = charm_dir
-        self._clients = {}  # client_id -> HookContext
-        self._invokers = {} # client_id -> Invoker
+        self._clients = {}   # client_id -> HookContext
+        self._invokers = {}  # client_id -> Invoker
 
         self._agent_log = logging.getLogger("unit-agent")
         self._agent_io = StringIO()
@@ -827,9 +827,13 @@
         # of the MESSAGE variable
         result = yield exe(self.get_test_hook("echo-hook"))
         self.assertEqual(result, 0)
-
         yield exe.ended
         self.assertIn(message, self.log.getvalue())
+
+        # Logging used to log an empty response dict
+        # assure this doesn't happpen  [b=915506]
+        self.assertNotIn("{}", self.log.getvalue())
+
         # The 'error' was sent via juju-log
         # to the UA. Our test UA has a fake log stream
         # which we can check now

